web/api/v1/api.go

```diff
@@ -889,6 +892,15 @@ func (api *API) serveFlags(r *http.Request) apiFuncResult {
 	return apiFuncResult{api.flagsMap, nil, nil, nil}
 }
 
+func (api *API) serveTSDBStatus(r *http.Request) apiFuncResult {
+	db := api.db()
+	if db == nil {
+		return apiFuncResult{nil, &apiError{errorUnavailable, errors.New("TSDB not ready")}, nil, nil}
+	}
+
+	return apiFuncResult{db.Head().PostingsCardinalityStats("__name__"), nil, nil, nil}
```

Comment:
I would suggest creating a new interface that will serve to serialize the data into JSON and potentially add more metadata if you need (like the `Target` struct in this file).